rocAL/source/pipeline/master_graph.cpp

```diff
@@ -440,12 +472,21 @@ MasterGraph::mem_type() {
 
 size_t
 MasterGraph::last_batch_padded_size() {
-    return _loader_module->last_batch_padded_size();
+    size_t max_last_batch_padded_size = 0;
+    for (auto loader_module : _loader_modules)
+        max_last_batch_padded_size = std::max(loader_module->last_batch_padded_size(), max_last_batch_padded_size);
+    return max_last_batch_padded_size;
 }
 
 Timing
 MasterGraph::timing() {
-    Timing t = _loader_module->timing();
+    Timing t;
+    // Accumulate the timings from each loader
+    for (auto loader_module : _loader_modules) {
+        Timing loader_time = loader_module->timing();
+        t.read_time = (t.read_time > loader_time.read_time) ? t.read_time : loader_time.read_time;
```

Comment:
Are we accumulating read time for each loader or setting the max read time here?

Reply:
we need to accumulate right?

Reply:
The above line checks the max read time and sets it to t. Shouldn't we use +=?

Reply:
Updated